code/modules/projectiles/guns/projectile/pistol.dm

```diff
@@ -157,29 +157,34 @@
 	fire_sound = 'sound/weapons/semiauto.ogg'
 	load_method = 2
 	gun_flags = SILENCECOMP | EMPTYCASINGS
+	silencer_offset = list(24,5)
 	var/obj/item/gun_part/glock_auto_conversion_kit/conversionkit = null
 
 /obj/item/weapon/gun/projectile/glock/update_icon()
 	..()
-	remove_overlays()
-	icon_state = "secglock[chambered ? "" : "-e"][silenced ? "-s" : ""][stored_magazine ? "" : "-m"][clowned == CLOWNED ? "-c" : ""]"
-	var/image/auto_overlay = image("icon" = 'icons/obj/biggun.dmi', "icon_state" = "auto_attach")
-	auto_overlay.pixel_x = chambered ? 0 : -3
-	if(conversionkit)
-		overlays += auto_overlay
-	
-/obj/item/weapon/gun/projectile/glock/proc/remove_overlays() //god this is HORRIBLE
-	var/image/auto_overlay = image("icon" = 'icons/obj/biggun.dmi', "icon_state" = "auto_attach")
-	auto_overlay.pixel_x = -3
-	overlays -= auto_overlay
-	auto_overlay.pixel_x = 0
-	overlays -= auto_overlay
+	if(!gun_part_overlays.len)
+		for(var/image/ol in gun_part_overlays)
```

Comment:
in order for this loop to to run `gun_part_overlays.len` needs to be 0, so this code will never do anything.

Reply:
🤔